osu.Game/Overlays/RestoreDefaultValueButton.cs

```diff
@@ -20,15 +20,30 @@ public class RestoreDefaultValueButton<T> : OsuButton, IHasTooltip, IHasCurrentV
     {
         public override bool IsPresent => base.IsPresent || Scheduler.HasPendingTasks;
 
-        private readonly BindableWithCurrent<T> current = new BindableWithCurrent<T>();
-
         // this is done to ensure a click on this button doesn't trigger focus on a parent element which contains the button.
         public override bool AcceptsFocus => true;
 
+        // this is intentionally not using BindableWithCurrent as this needs a BindableNumber instance for an accurate IsDefault value.
+        //
+        // this also cannot use IBindableWithCurrent.Create() due to BindableNumberWithCurrent
+        // directly casting given bindables to BindableNumber, which is not necessarily the case.
+        //
+        // therefore rely on the old method of taking each current bindable instance for now, until things are settled framework-side.
+        private Bindable<T> current;
+
         public Bindable<T> Current
         {
-            get => current.Current;
-            set => current.Current = value;
+            get => current;
+            set
+            {
+                current?.UnbindAll();
```

Comment:
It sure looks like `UnbindAll` won't unregister from `DefaultChanged` to me (see [link 1](https://github.com/ppy/osu-framework/blob/d6d14ce4723def41e5b4af7216824752882b4d49/osu.Framework/Bindables/Bindable.cs#L363-L370), [link 2](https://github.com/ppy/osu-framework/blob/d6d14ce4723def41e5b4af7216824752882b4d49/osu.Framework/Bindables/Bindable.cs#L335-L339)).

Reply:
That looks like a very bad oversight... don't think anything would've unbound them and references were leaked for those which used them all the time.

Reply:
This should be fine to go as this was the case previously anyways, and this doesn't worsen anything in that aspect.